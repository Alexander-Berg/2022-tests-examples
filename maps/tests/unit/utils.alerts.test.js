'use strict';

const {generateGolovanAppAlert, generateGolovanL7Alert} = require('../../lib/utils/alerts');

const qtoolsConfig = require('./fixtures/qtools-for-alerts.json');

describe('utils/alerts', () => {
    describe('#generateGolovanAppAlert()', () => {
        qtoolsConfig.geoSpecific.monitorings.components.forEach((component) => {
            it(`should make correct generation for component ${component.environment}/${component.name}`, () => {
                const golovanAlert = generateGolovanAppAlert({
                    fullEnvName: `${qtoolsConfig.abcServiceName}_${component.environment}`,
                    component,
                    maintainers: ['robot-ololo']
                });

                expect(mockQtoolsVersion(golovanAlert)).toMatchSnapshot();
            });
        });
    });

    describe('#generateGolovanL7Alert()', () => {
        qtoolsConfig.geoSpecific.monitorings.l7.forEach((balancer) => {
            it(`should make correct generation for balancer ${balancer.id}`, () => {
                const golovanAlert = generateGolovanL7Alert({
                    balancer,
                    maintainers: ['robot-ololo']
                });

                expect(mockQtoolsVersion(golovanAlert)).toMatchSnapshot();
            });
        });
    });
});

function mockQtoolsVersion(alert) {
    return alert.replace(/<% set qtools_version = '(.*)' %>/, '<% set qtools_version = \'TBD\' %>');
}
