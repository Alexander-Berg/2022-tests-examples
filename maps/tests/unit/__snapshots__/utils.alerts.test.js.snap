// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/alerts #generateGolovanAppAlert() should make correct generation for component production/app 1`] = `
"<% import 'maps-front-common-application-alerts' as alerts %>
<% set qtools_version = 'TBD' %>
<%
    set full_env_name = 'qtools-test-service_production'
%>
<%
    set component_name = 'app'
%>
<%
    set workload_name = 'app_workload'
%>
<%
    set maintainers = [\\"robot-ololo\\"]
%>
<%
    set access_checks = {
    \\"total\\": {
        \\"vhost-2xx-drop\\": {
            \\"warn\\": [
                0,
                10
            ],
            \\"crit\\": [
                10,
                None
            ]
        },
        \\"vhost-2xx-trend-down\\": {
            \\"warn_perc\\": 10,
            \\"crit_perc\\": 20
        },
        \\"vhost-4xx\\": {
            \\"warn\\": [
                5,
                10
            ],
            \\"crit\\": [
                10,
                None
            ]
        },
        \\"vhost-429\\": {
            \\"warn\\": [
                5,
                10
            ],
            \\"crit\\": [
                10,
                None
            ]
        },
        \\"vhost-499\\": {
            \\"warn\\": [
                5,
                10
            ],
            \\"crit\\": [
                10,
                None
            ]
        },
        \\"vhost-5xx\\": {
            \\"warn\\": [
                1,
                10
            ],
            \\"crit\\": [
                10,
                None
            ]
        }
    }
}
%>
<%
    set alerts_checks = {
    \\"unispace\\": {
        \\"warn\\": [
            90,
            95
        ],
        \\"crit\\": [
            95,
            None
        ]
    }
}
%>
<%
    set app_checks = {}
%>
<%
    set alerts_raw_checks = {
    \\"total_4xx\\": {
        \\"signal\\": \\"unistat-roquefort-ex-ping_total_4xx_ammv\\",
        \\"warn\\": [
            5,
            10
        ],
        \\"crit\\": [
            10,
            None
        ],
        \\"flaps\\": {
            \\"boost\\": 0,
            \\"stable\\": 300,
            \\"critical\\": 1200
        },
        \\"name\\": \\"total_4xx\\"
    }
}
%>
[<< alerts.projectChecks(
   full_env_name=full_env_name,
   component_name=component_name,
   access_checks=access_checks,
   alerts_checks=alerts_checks,
   app_checks=app_checks,
   alerts_raw_checks=alerts_raw_checks,
   maintainers=maintainers,
   workload_name=workload_name
)
>>]"
`;

exports[`utils/alerts #generateGolovanL7Alert() should make correct generation for balancer first-qtools-test-service.slb.maps.yandex.net 1`] = `
"<% import 'maps-front-common-application-alerts' as alerts %>
<% set qtools_version = 'TBD' %>
<%
    set balancer_checks = {
    \\"id\\": \\"first-qtools-test-service.slb.maps.yandex.net\\",
    \\"infra\\": {
        \\"cpu-usage\\": {
            \\"warn\\": [
                40,
                60
            ],
            \\"crit\\": [
                60,
                None
            ]
        },
        \\"cpu-wait\\": {
            \\"warn\\": [
                0.2,
                0.5
            ],
            \\"crit\\": [
                0.5,
                None
            ]
        },
        \\"memory-usage\\": {
            \\"warn\\": [
                85,
                95
            ],
            \\"crit\\": [
                95,
                None
            ]
        }
    },
    \\"upstreams\\": {
        \\"main\\": {}
    }
}
%>
<%
    set maintainers = [\\"robot-ololo\\"]
%>
[<< alerts.slbChecks(
   balancer_checks=balancer_checks,
   maintainers=maintainers
)
>>]"
`;

exports[`utils/alerts #generateGolovanL7Alert() should make correct generation for balancer second-qtools-test-service.slb.maps.yandex.net 1`] = `
"<% import 'maps-front-common-application-alerts' as alerts %>
<% set qtools_version = 'TBD' %>
<%
    set balancer_checks = {
    \\"id\\": \\"second-qtools-test-service.slb.maps.yandex.net\\",
    \\"infra\\": {},
    \\"upstreams\\": {
        \\"main\\": {
            \\"inprogress-connection\\": {
                \\"warn\\": [
                    11,
                    16
                ],
                \\"crit\\": [
                    16,
                    None
                ]
            },
            \\"fail-connection\\": {
                \\"warn\\": [
                    2.4,
                    4.5
                ],
                \\"crit\\": [
                    4.5,
                    None
                ]
            },
            \\"connection-timeout\\": {
                \\"warn\\": [
                    2.3,
                    5.6
                ],
                \\"crit\\": [
                    5.6,
                    None
                ]
            },
            \\"backend-timeout\\": {
                \\"warn\\": [
                    11,
                    19
                ],
                \\"crit\\": [
                    19,
                    None
                ]
            },
            \\"connection-refused\\": {
                \\"warn\\": [
                    1,
                    31
                ],
                \\"crit\\": [
                    31,
                    None
                ]
            },
            \\"vhost-5xx\\": {
                \\"warn\\": [
                    41,
                    68
                ],
                \\"crit\\": [
                    68,
                    None
                ]
            }
        }
    }
}
%>
<%
    set maintainers = [\\"robot-ololo\\"]
%>
[<< alerts.slbChecks(
   balancer_checks=balancer_checks,
   maintainers=maintainers
)
>>]"
`;

exports[`utils/alerts #generateGolovanL7Alert() should make correct generation for balancer third-qtools-test-service.slb.maps.yandex.net 1`] = `
"<% import 'maps-front-common-application-alerts' as alerts %>
<% set qtools_version = 'TBD' %>
<%
    set balancer_checks = {
    \\"id\\": \\"third-qtools-test-service.slb.maps.yandex.net\\",
    \\"infra\\": {
        \\"cpu-wait\\": {
            \\"warn\\": [
                0.3,
                0.45
            ],
            \\"crit\\": [
                0.45,
                None
            ]
        },
        \\"memory-usage\\": {
            \\"warn\\": [
                87,
                92
            ],
            \\"crit\\": [
                92,
                None
            ]
        }
    },
    \\"upstreams\\": {
        \\"main\\": {
            \\"inprogress-connection\\": {
                \\"warn\\": [
                    13,
                    23
                ],
                \\"crit\\": [
                    23,
                    None
                ]
            }
        },
        \\"additional\\": {
            \\"vhost-5xx\\": {
                \\"warn\\": [
                    44,
                    57
                ],
                \\"crit\\": [
                    57,
                    None
                ]
            }
        }
    }
}
%>
<%
    set maintainers = [\\"robot-ololo\\"]
%>
[<< alerts.slbChecks(
   balancer_checks=balancer_checks,
   maintainers=maintainers
)
>>]"
`;
